lib/syntax_tree/node.rb

```diff
@@ -9324,6 +9324,48 @@ def ambiguous?(q)
     end
   end
 
+  # RemovedNode is a blank node used in places of nodes that have been removed.
+  class RemovedNode < Node
+    # [Array[ Comment | EmbDoc ]] the comments attached to this node
+    attr_reader :comments
+
+    def initialize(location:)
```

Comment:
I wasn't sure about exact API of the `RemovedNode`. I think we may want `RemovedNode` to point to an instance of the node it substitutes but I at the moment I don't have use-cases for it so I decided not to overcomplicate. Perhaps I should have simplified it even further and dropped the `comments` and `location` properties but some existing code expected every node to respond to `comments` so I decided to avoid breaking this expectation